lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -219,10 +213,15 @@ private JourneyResponse getJourneyResponse(
             final boolean isF2FIncomplete = !Objects.isNull(f2fRequest) && !hasF2fVc;
             final boolean isF2FComplete = !Objects.isNull(f2fRequest) && hasF2fVc;
 
-            var ciScoringCheckResponse =
+            Optional<JourneyResponse> ciScoringCheckResponse =
                     checkForCIScoringFailure(
                             ipAddress, clientOAuthSessionItem, govukSigninJourneyId);
 
+            Boolean reproveIdentity = clientOAuthSessionItem.getReproveIdentity();
+            if ((reproveIdentity != null && reproveIdentity)
+                    || configService.enabled(RESET_IDENTITY.getName())) {
+                return buildForceResetResponse(ciScoringCheckResponse.orElse(null));
```

Comment:
Can we just pass the optional to this method, rather than falling back to nullable types?